### PR TITLE
Remove CastleMonoRail2 instrumentation.

### PR DESCRIPTION
**Pull request by vuqtran88**
## Description
Resolves [#702 ]

### CODE REVIEW

**Review by JcolemanNR — CHANGES_REQUESTED** [review]
Looks good, but since this a deprecation (similar to the logging instrumentation deprecation), we should do the following:

- [ ] Add the instrumentation XML to the instrumentation ignore list in the profiler.
- [ ] Attempt to remove the DLL/XML in the MSI
- [ ] Attempt to remove the DLL/XML in the Azure Site Extension (I think there is prior art)

**Comment by vuqtran88** [comment]
@JcolemanNR I added the change as requested, except for the Azure Site Extension. With the change in the profiler, the risk is nearly 0 so i don't think it is worth to do the update.

File: src/Agent/CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Deprecations/Removed Features
 * The scriptable installers have been removed. ([#1170](https://github.com/newrelic/newrelic-dotnet-agent/pull/1170))
+* Castle.Monorail v2 instrumentation has been reomoved. ([#1177])(https://github.com/newrelic/newrelic-dotnet-agent/pull/1177)
```
Comment (nr-ahemsath) [inline src/Agent/CHANGELOG.md:17]:
```suggestion
* Castle.Monorail instrumentation has been removed. ([#1177])(https://github.com/newrelic/newrelic-dotnet-agent/pull/1177)
```

I think that by specifying "v2" specifically, somebody might think that we still have instrumentation for other versions of Castle.Monorail, when we don't.
```suggestion
* Castle.Monorail v2 instrumentation has been reomoved. ([#1177])(https://github.com/newrelic/newrelic-dotnet-agent/pull/1177)
```

**Review by JcolemanNR — APPROVED** [review]
As long as the profiler is rebuilt/checked in before this merges to main then I'm fine with thig going in as is without the profiler binaries

**Review by nr-ahemsath — APPROVED** [review]
Looks good to me.  Looks like all review feedback has been handled.  👍

**Pull request merged**